frontend/src/components/Wallet.js

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: MIT
+// SDPX-FileCopyrightText: 2021 Que Le <b.le@tu-berlin.de>
+
+
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+const { utils } = require( "ethers" );
+import { delayMsec } from "../utils/utils";
+
+import Typography from "@mui/material/Typography";
+import {
+	Button,
+	Container,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import Grid from "@mui/material/Grid";
+import ConditionalAlert from "./ConditionalAlert";
+
+function AccountEntry({ data }) {
+	let balance = "NaN";
+	// data.balance can have NaN value if no Metamask or other Wallet installed
+	if (!isNaN(data.balance)) {
+		balance = utils.formatEther(data.balance);
+	}
+	
+	return <TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{balance}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthereumAccountDetails ({ publicAddresses, setErrorNoMetamaskMessage }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		// TODO: publicAddresses is still not loaded. 
```

Comment:
Here, check if publicAddresses is loaded and if not just return or throw an exception. Below, in the useEffect, add `publicAddresses` in the square brackets. The function in useEffect is called every time any of the states in the square brackets changes. Thus, whenever publicAddresses, this function will be called. Contact me if you would like some more clarifications.

Reply:
this could have reduced around 3 hours working for me. How could I oversee this ^^